packages/service/ni_measurement_plugin_sdk_service/_internal/parameter/serialization_descriptors.py

```diff
@@ -1,76 +1,43 @@
 """Serialization Descriptors."""
 
 from json import loads
-from typing import Any, List
+from typing import List
 
 from google.protobuf import descriptor_pb2, descriptor_pool
-from google.protobuf.descriptor_pb2 import FieldDescriptorProto
+from google.protobuf.descriptor_pb2 import DescriptorProto, FieldDescriptorProto
 
 from ni_measurement_plugin_sdk_service._annotations import ENUM_VALUES_KEY
+from ni_measurement_plugin_sdk_service._internal.parameter._get_type import (
+    _TYPE_FIELD_MAPPING,
+)
 from ni_measurement_plugin_sdk_service._internal.parameter.metadata import (
     ParameterMetadata,
 )
-from tests.utilities.stubs.loopback.types_pb2 import ProtobufColor
-
-
-def _get_output_enum_type(
-    metadata_enum_list: List[str],
-    file_descriptor: descriptor_pb2.FileDescriptorProto,
-) -> Any:
-    """Get's matching enum class from 'file_descriptor'.
 
-    Args:
-        metadata_enum_list (List[str]): Enum names from metadata.annotations.
 
-        file_descriptor: Descriptor of proto file.
-
-    Returns:
-        Any: Matching enum class in a str type or None when enum is protobuf.
-    """
-    for enum_type in file_descriptor.enum_type:
-        enum_names = [enum_value.name for enum_value in enum_type.value]
-        if sorted(metadata_enum_list) == sorted(enum_names):
-            return enum_type.name
-    return None
-
-
-def _create_enum_type(
+def _create_enum_type_class(
     file_descriptor: descriptor_pb2.FileDescriptorProto,
     parameter_metadata: ParameterMetadata,
     field_descriptor: FieldDescriptorProto,
 ) -> None:
-    """Implement a enum class in 'file_descriptor'.
-
-    Args:
-        file_descriptor (FileDescriptorProto): Descriptor of a proto file.
-
-        parmeter_metadata (ParameterMetadata): Metadata of current field.
-
-        field_descriptor (FieldDescriptorProto): Descriptor of a field.
-
-    Returns:
-        None: Only creates a enum class in file_descriptor.
-    """
+    """Implement a enum class in 'file_descriptor'."""
     enum_dict = loads(parameter_metadata.annotations[ENUM_VALUES_KEY])
-    if parameter_metadata.default_value is None:
-        enum_type_name = _get_output_enum_type(
-            metadata_enum_list=[enum for enum in enum_dict.keys()], file_descriptor=file_descriptor
-        )
-    else:
-        enum_type_name = _get_enum_type(parameter_metadata).__name__
+    enum_type_name = _get_enum_type(parameter_metadata).__name__
+
+    # if enum is a protobuf then enum_type_name is 1st letter of each enum name
+    # e.g. {"NONE": 0, "RED": 1, "GREEN": 2} -> NRG
+    if enum_type_name == "int" or enum_type_name == "NoneType":
```

Comment:
Why does this check for both int and NoneType? Protobuf doesn't use both of these, does it? The comment says we do this for protobuf enums but are we doing it for some other case too?

Reply:
Int is what I was getting for the type of the default value since the default values for protobuf enums are a integer, not a enum. Same thing for None since the default value for output values is None for some reason.

I just realized this condition catches both python and protobuf output enums, which I somehow thought it was ok to do. I'm not planning to use this in the next commit so this is basically resolved.